fix: update code to use federation and gateway plugins

ekynoxe opened the pull request:
fixes #117 
fixes #125

ekynoxe commented:
I probably need to update the README as well.

simoneb commented:
since the upgrade slipped in without us noticing, do you think there's a way to write a test that would have caught the breakage?

simoneb commented:
If the example was the only thing affected, then we can get away without the test

ekynoxe commented:
Yes, the tests were already using @mercurius/gateway: https://github.com/nearform/mercurius-federation-info/blob/master/test/basic.test.js#L42

I've also updated the README to avoid confusion.

ekynoxe commented:
Actually, @codeflyer @simoneb thinking about it, would we still not have to add a check in https://github.com/nearform/mercurius-federation-info/blob/master/index.js#L21 for `fastify.graphqlGateway` being defined to avoid crashes just like we're seeing in the example server and in other plugins like #117  (which was raised from https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/235)

How can we ensure that with mercurius 12 is used **with** the `@mercurius/gateway` plugin?

simoneb commented:
I don't remember if there's a reason why we didn't do it, but since this is a fastify plugin and it depends on other plugins to have been registered, shall we not use a plugin dependency then? See https://github.com/fastify/fastify-plugin#dependencies

ekynoxe commented:
@codeflyer I've added a test and also found another issue with the readme that I've fixed.
I wanted to test the logger output, but being green to `tap` I couldn't find an easy way to mock it and assert a single call to check for the "gateway not present" message (there are other logs that would need to be ignored, and I can't find a way to do what jest would offer with its `toHaveBeenCalledWith` assertion)

I think we should be good to go with this last check. Once merged, we can deploy it and fix https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/235

codeflyer commented on test/basic.test.js:
```diff
@@ -8,6 +9,70 @@ import { readFileSync } from 'fs'
 const fileUrl = new URL('../package.json', import.meta.url)
 const packageJSON = JSON.parse(readFileSync(fileUrl))
 
+test('"/federation-schema" returns a 404 error if fastify is intanciated without the gateway', async t => {
```
typo: intanciated  -> instantiated

ekynoxe commented:
Thanks @codeflyer I've implemented the first solution to check the log and fixed the typo.

codeflyer commented on test/basic.test.js:
```diff
@@ -8,6 +9,81 @@ import { readFileSync } from 'fs'
 const fileUrl = new URL('../package.json', import.meta.url)
 const packageJSON = JSON.parse(readFileSync(fileUrl))
 
+test('"/federation-schema" returns a 404 error if fastify is intantiated without the gateway, and logs an error message', async t => {
```
still missing the `s` -> inStantiated

ekynoxe replied:
Argh, sorry about that. 

Fixed (and I've installed a spell checker too!)

codeflyer reviewed (approved):
LGTM
